dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Users.java

```diff
@@ -0,0 +1,47 @@
+package edu.harvard.iq.dataverse.api;
+
+import edu.harvard.iq.dataverse.persistence.user.User;
+import edu.harvard.iq.dataverse.users.ChangeUserIdentifierService;
+
+import javax.ejb.EJBException;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+/**
+ *
+ * @author madunlap
+ */
+@Path("users")
+public class Users extends AbstractApiBean {
+
+    @Inject
+    private ChangeUserIdentifierService changeUserIdentifierService;
+
+    @POST
+    @Path("{identifier}/changeIdentifier/{newIdentifier}")
+    public Response changeAuthenticatedUserIdentifier(@PathParam("identifier") String oldIdentifier, @PathParam("newIdentifier")  String newIdentifier) {
+        try {
+            User user;
+            user = findUserOrDie();
+            changeUserIdentifierService.changeUserIdentifier(user, oldIdentifier, newIdentifier);
+
+            return ok("UserIdentifier changed from " + oldIdentifier + " to " + newIdentifier);
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        } catch (SecurityException se) {
+            return error(Response.Status.UNAUTHORIZED, se.getLocalizedMessage());
+        } catch (IllegalArgumentException iae) {
+            return error(Response.Status.EXPECTATION_FAILED, iae.getLocalizedMessage());
+        } catch (IllegalStateException ise) {
+            return error(Response.Status.EXPECTATION_FAILED, ise.getLocalizedMessage());
```

Comment:
I don,t think that there is possibility to reach catch for IllegalArgumentException, State or SecurityException. They will be wrapped by EjbException